src/coap_gnutls.c

```diff
@@ -2349,6 +2350,39 @@ ssize_t coap_tls_read(coap_session_t *c_session,
 }
 #endif /* !COAP_DISABLE_TCP */
 
+coap_digest_ctx_t *
+coap_digest_setup(void) {
+  gnutls_hash_hd_t digest_ctx;
+
+  if (gnutls_hash_init(&digest_ctx, GNUTLS_DIG_SHA256)) {
+    return NULL;
+  }
+  return digest_ctx;
+}
+
+void
+coap_digest_free(coap_digest_ctx_t *digest_ctx) {
+  gnutls_hash_deinit(digest_ctx, NULL);
+}
+
+int
+coap_digest_update(coap_digest_ctx_t *digest_ctx,
+                   const uint8_t *data,
+                   size_t data_len) {
+  int ret = gnutls_hash(digest_ctx, data, data_len);
+
+  return ret == 0 ? 1 : 0;
```

Comment:
`return ret == 0;` seems clearer to me, using the implicit conversion from bool to int.

Reply:
updated.